### DISCUSSION TITLE
Web UI doesn't recognize licenses that comply with REUSE

### DISCUSSION THREAD

**Discussion by kyper-james** [community/community#15033] [Other Feature Feedback, Questions, & Ideas]
Hi all,

I've recently updated the copyright and licensing metadata in some of my projects to comply with the [REUSE](https://reuse.software) specification (which has been partially adopted by the Linux kernel, for example). However, GitHub's web UI doesn't recognize the licenses in the mandated ```LICENSES``` directory, and therefore doesn't link to them in the ```About``` panel on the right-hand side of the project page.

See [https://github.com/kyper-io/alfa](https://github.com/kyper-io/alfa) as an example:

![Screenshot from 2022-04-16 09-44-23](https://user-images.githubusercontent.com/68329770/163668605-b1cc6c3c-755e-4d70-a22a-2339342e6d3b.png)

This can be worked around by adding a ```COPYRIGHT``` or ```COPYING``` file (despite it not being required by [REUSE](https://reuse.software)), but it would be more convenient if GitHub could detect it automatically and streamline the licensing process even further.

What do you think?

**Comment by sdavids** [comment]
Duplicate of #44922

**Reply by sdruskat** [reply]
> Duplicate of #44922

@sdavids I think this is slightly misleading :). This discussion here is the much older one, and I think should be understood as being the "original", of which #44922 is the duplicate.

Right now, both state that they are a duplicate of each other, so perhaps this is one of the reasons why neither of the two discussions get much attention?

**Comment by sdruskat** [comment]
It would be great to have the web UI support REUSE-specified licenses!

I understand it may be complicated, as the UI is meant to show only the "one true license" (which isn't something that arguably exists, given the different types of files in a repo).

Some approaches to solving this would be, e.g.:

- Show a "REUSE-compliant", "Specified using REUSE" or similar test (or badge?) instead of a license name, if
    - the currently viewed page shows a state of the repo for which a workflow running the [REUSE action](https://reuse.software/dev/#github) ([repo](https://github.com/fsfe/reuse-action)) has passed (stronger validation), or
    - a `REUSE.toml` file exists (less strong validation), or
    - a `LICENSES` folder exists containing only `.txt` files that are named after valid SPDX license identifiers (e.g., `MIT.txt`, `CC-BY-SA-4.0.txt`, etc.) (stronger but more complicated validation)
